### PR TITLE
Bug 1732579: WIP: Permissive/Strict Options

### PR DESCRIPTION
**Pull request by njhale**
- Stores all load errors encountered instead of exiting early.
- Adds a `permissive` option to the `intializer` and `configmap-server` commands that, when given, permits load errors with out failing.
- Adds a `strict` option to the `appregistry-server` command that, when given, fails when load errors are encountered.

Supersedes #57

### CODE REVIEW

File: pkg/registry/errors.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	if e.errs == nil {
+		e.errs = map[LoadErrorType][]LoadError{}
+	}
+	if e.typeRegistry == nil {
+		e.typeRegistry = map[LoadErrorType]reflect.Type{}
+	}
+
+	underlyingType := reflect.TypeOf(err.Err)
+	if registeredType, registered := e.typeRegistry[err.Type]; registered && underlyingType != registeredType {
+		return errors.Errorf("underlying error type %s doesn't match registered type %s for %s", underlyingType.Name(), registeredType.Name(), err.Type)
```
Comment (ecordell) [inline pkg/registry/errors.go]: Should this panic as a programmer error?

File: pkg/appregistry/appregistry.go [hunk 2]
```diff
@@ -67,7 +67,18 @@ type AppregistryLoader struct {
 	loader     *dbLoader
 }
 
-func (a *AppregistryLoader) Load(csvSources []string, csvPackages string) (store registry.Query, err error) {
+func (a *AppregistryLoader) Load(csvSources []string, csvPackages string) (registry.Query, error) {
+	store, err := a.Load(csvSources, csvPackages)
```
Comment (ecordell) [inline pkg/appregistry/appregistry.go]: Should this be `a.load`?

Comment (njhale) [reply pkg/appregistry/appregistry.go]: yup - I just pushed before I left the office

**Comment by njhale** [comment]
Did some manual testing with a bad app-registry repo:
```
# permissive default behavior
$ go run ./cmd/appregistry-server/main.go -r https://quay.io/cnr\|njhale --kubeconfig ~/.kube/config -o etcd
<elided download logs>
WARN[0001] permitted load errors:[package etcd does not define a default channel, package/channel etcd/singlenamespace-alpha csv etcdoperator.v0.9.2 specifies replaces that could not be found: etcdoperator.v0.9.1, package/channel etcd/clusterwide-alpha csv etcdoperator.v0.9.0 specifies replaces that could not be found: etcdoperator.v0.8.2]  port=50051 type=appregistry
INFO[0001] serving registry                              port=50051 type=appregistry

# optional strict behavior
$ go run ./cmd/appregistry-server/main.go -r https://quay.io/cnr\|njhale --kubeconfig ~/.kube/config -o etcd --strict
<elided download logs>
FATA[0000] strict mode enabled, failing on load errors:
[package etcd does not define a default channel, package/channel etcd/singlenamespace-alpha csv etcdoperator.v0.9.2 specifies replaces that could not be found: etcdoperator.v0.9.1, package/channel etcd/clusterwide-alpha csv etcdoperator.v0.9.0 specifies replaces that could not be found: etcdoperator.v0.8.2]  port=50051 type=appregistry
exit status 1
```

**Pull request closed without merging**